internal/connor/eq.go

```diff
@@ -29,28 +29,32 @@ func eq(condition, data any) (bool, error) {
 		return false, nil
 
 	case immutable.Option[bool]:
-		if !arr.HasValue() {
-			return condition == nil, nil
+		if arr.HasValue() {
+			data = arr.Value()
+		} else {
+			data = nil
 		}
-		data = arr.Value()
 
 	case immutable.Option[int64]:
-		if !arr.HasValue() {
-			return condition == nil, nil
+		if arr.HasValue() {
+			data = arr.Value()
+		} else {
+			data = nil
 		}
-		data = arr.Value()
 
 	case immutable.Option[float64]:
-		if !arr.HasValue() {
-			return condition == nil, nil
+		if arr.HasValue() {
+			data = arr.Value()
+		} else {
+			data = nil
 		}
-		data = arr.Value()
 
 	case immutable.Option[string]:
-		if !arr.HasValue() {
-			return condition == nil, nil
+		if arr.HasValue() {
+			data = arr.Value()
+		} else {
+			data = nil
```

Comment:
suggestion: this could also be turned into generic code

Reply:
done